src/originalBudgetTrackingApp/index.ts

```diff
@@ -21,64 +22,66 @@ const DATE_FORMAT = 'DD/MM/YYYY';
 
 export const { getYnabAccountDetails } = ynab;
 
-export async function scrapeAndUpdateOutputVendors() {
+export async function scrapeAndUpdateOutputVendors(optionalEventEmitter?: ScrapingEventEmitter) {
+  const eventEmitter = new EmptyEventEmitterAdapter(optionalEventEmitter);
```

Comment:
If you keep the `ScrapingEventEmitter` as `nullable`, so you have to pass it as `nullable` to all methods, and I was afraid the if some method declaring it as `nullable`, you may forget to move it within out internal implementation, and there will be no error.

To solve this I declared it as `nullable` only in the `export`ed method, then I adapted it to `EmptyEventEmitterAdapter`, and then I can pass it as not null, even if it null.

Not sure it is a good practice at all.